RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,301 @@
+This mechanism can be used to perform private discovery requests, where the online communication address of the source node is encrypted and propagated through the network until it reached the destination node.
+Private discovery requests can only be performed if both parties are online at the same time.
+Encryption of the data message ensuring that only the destination node is able to view the online address of the source node as the data message moves through the network.
+Once the destination node receives and decrypts the data message, that node is then able to establish a P2P communication channel with the source node for any further communication.
+
+Propagation of completely private discovery request, hidden as an encrypted data message, can be performed as a broadcast through the entire network using the Gossip protocol.
+Propagation of public discovery requests can be performed using more efficient directed propagation using the Kademlia protocol.
+As encrypted message with visible destinations tend to not be of interest to the rest of the network, directed propagation using the Kademlia protocol to forward these messages to the correct parties are preferred.
+
+This same encryption technique can be used to send encrypted messages to a single node or a group of nodes, where the group of nodes have shared identification keys.
+A Validation Committee is an example of a group of CNs that have shared identification keys for the committee.
+The shared identification keys ensure that all members of that committee are able to receive and decrypt data messages that were sent to the committee.
+
+#### Maintaining connections with peers
+
+CCs and CNs establish and maintain connections with peers differently.
+CCs only create a few short-lived ad hoc channels and CNs create and maintain long-lived channels with a number of peers.
+
+If a CC is unaware of a destination CNs or CCs online communication address then the address first needs to be obtained using a discovery request.
+When a CC already know the communication address of the CC or CN that he wants to communicate with, then a direct P2P channel can be established between the two peers for the duration of the communication task.
+The communication channel can then be closed as soon as the communication task has been completed.
+
+CNs consisting of VNs and BNs typically attempt to maintain communication channels with a large number of peers.
+The distribution of peers (VNs vs BNs) that a single CN keeps communication channels open with can change depending on the type of node.
+A CN that is also a BN should maintain more peer connections with other BNs, but should also have some connections with other VNs.
+
+Having some connections with VNs are important as BNs have derived node ids and not registered node ids such as VNs, making it possible for the CN to be separated from the main network and become victim to an eclipse attack.
```

Comment:
add a ref link explaining eclipse attacks eg: https://www.radixdlt.com/post/what-is-an-eclipse-attack

Reply:
Added a link to a paper on Eclipse attack at first use, also added your suggested link to second use.